Bluepill-cli/bp_test_dumper/bp_test_dumper.m

```diff
@@ -0,0 +1,90 @@
+//
+//  bp_test_dumper.m
+//  bp_test_dumper
+//
+//  Created by Ravi K. Mandala on 10/23/19.
+//  Copyright © 2019 LinkedIn. All rights reserved.
+//
+#import <Foundation/Foundation.h>
+#import <XCTest/XCTest.h>
+
+NSString *getSimpleName(NSString *testName) {
+    // Extracting test name from a string like "-[FeedHighlightedUpdateTrackingTest testFeedHighlightedUpdateEvent]"
+    NSString *simpleTestName = [testName componentsSeparatedByString:@" "][1];
+    simpleTestName = [simpleTestName substringToIndex:simpleTestName.length-1];
+    return simpleTestName;
+}
+
+/* Return code:
+ * 0 - Successful
+ * 1 - Error loading the test bundles
+ * 2 - Error writing output file
+ */
+__attribute__((constructor))
+void read_it() {
+    NSString *outputFilePath = [[NSProcessInfo processInfo] environment][@"_BP_TEST_LIST_FILE"] ?: @"/tmp/test_list.json";
```

Comment:
Please don't hard code the output file in the `else` case. I think if the env variable is not set this should fail no, if you want to still produce output use a temp file via some temp file API. If you have reasons for not wanting to use a temp file API, at least put the PID of the app in the file name.

Reply:
It should better fail if this environment variable is not set. Because if we use an alternate file name it has to be looked for and read, without which writing an alternate file is meaningless. I will make it fail.